### DISCUSSION TITLE
/disable-autofocus-in-octoprint-with-a-logitech-c920-webcam/

### DISCUSSION THREAD

**Discussion by migratecomments** [ironicbadger/blog.ktz.me#10] [Announcements]
Disable autofocus in Octoprint with a Logitech C920 webcam
        
[https://blog.ktz.me/disable-autofocus-in-octoprint-with-a-logitech-c920-webcam/](https://blog.ktz.me/disable-autofocus-in-octoprint-with-a-logitech-c920-webcam/)

**Comment by migratecomments** [comment]
```header
Originally posted by OriginStarSeeker on 2020-12-14 09:21
```

Any reason not to just switch the settings here?

http://octopi.local:8080/control.htm

**Reply by migratecomments** [reply]
```header
Originally posted on 2020-12-14 14:27
```

How to enable this? I get:

404: Not Found!  
Could not open file

**Reply by migratecomments** [reply]
```header
Originally posted by OriginStarSeeker on 2020-12-14 14:48
```

Odd. It should come on octopi by default. You can get to octoprint at https://octopi.local?

Does the camera show up there?

**Reply by migratecomments** [reply]
```header
Originally posted on 2021-03-29 22:02
```

See this thread, the feature was disabled by default for security concerns.

[https://community.octoprint...](https://community.octoprint.org/t/cant-access-control-htm/19050/2)

**Comment by migratecomments** [comment]
```header
Originally posted by Kevan Dalby on 2022-02-24 04:40
```

Hi, I have a C920 and a C270 with a Rasp Pi cam all running on a Pi4. How would I go about disabling the auto focus on the two Logitech cameras?? Thanks

**Comment by migratecomments** [comment]
```header
Originally posted by leonardwinck on 2022-03-10 03:49
```

Fantastic guide, thank you! Managed to quickly set the focus and not have any more focus pumping in my octolapse recordings.  
One thing I'd like to mention though:

`sudo v4l2-ctl --set-ctrl=focus_auto=0   sudo v4l2-ctl --set-ctrl=focus_absolute=32`

The "sudo" can be omitted here when logged in as user pi or root, which is the default.  
Executing commands with sudo which do not need to be executed with sudo introduces security risks and should be avoided. Leads to a bad habit of just executing everything with sudo way too often.

**Comment by bohaman1** [comment]
Hello,

When I tried the command "sudo v4l2-ctl --set-ctrl=focus_auto=0"

I've received the error "unknown command" I have already installed v4l2. May I ask what I could have done wrong ?

**Reply by bohaman1** [reply]
Sorry the exact error is "unknown control 'focus_auto'"
